### PR TITLE
Jenkins longevity abort callback

### PR DESCRIPTION
**Pull request by dkropachev**
https://trello.com/c/Hgy9sdJC/1753-send-longevity-email-in-any-case
Should be merged after https://github.com/scylladb/scylla-cluster-tests/pull/2145

## PR pre-checks (self review)

- [x] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [x] I didn't leave commented-out/debugging code
- [x] I added the relevant `backport` labels
- [ ] ~~New configuration option are added and documented (in `sdcm/sct_config.py`)~~
- [ ] ~~I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)~~
- [ ] ~~All new and existing unit tests passed (CI)~~
- [ ] ~~I have updated the Readme/doc folder accordingly (if needed)~~

### CODE REVIEW

**Comment by dkropachev** [comment]
These were two tests done on this pipeline:

General failure test - https://jenkins.scylladb.com/view/master/job/scylla-master/job/longevity/job/longevity-mv-si-4days-test-test/16/

Abortion test - https://jenkins.scylladb.com/view/master/job/scylla-master/job/longevity/job/longevity-mv-si-4days-test-test/18/

File: vars/longevityPipeline.groovy [hunk 1]
```diff
@@ -51,6 +170,11 @@ def call(Map pipelineParams) {
             string(defaultValue: "${pipelineParams.get('email_recipients', 'qa@scylladb.com')}",
                    description: 'email recipients of email report',
                    name: 'email_recipients')
+
+            string(defaultValue: "${pipelineParams.get('test_config', '')}",
```
Comment (fruch) [inline vars/longevityPipeline.groovy:174]: can we not expose this value ?
also seems like `test_name` isn't part of params

File: vars/longevityPipeline.groovy [hunk 2]
```diff
@@ -1,5 +1,124 @@
 #!groovy
 
+def runSctTest(Map params){
+    // handle params which can be a json list
+    params = params.params
```
Comment (fruch) [inline vars/longevityPipeline.groovy:5]: if pipelineParams is passed to this function, I can understand this, but this isn't the case

Comment (fruch) [reply vars/longevityPipeline.groovy:5]: now I get it, since a named param is used, this is very confusing...

Comment (dkropachev) [reply vars/longevityPipeline.groovy:5]: I tried hard, but it looks like there is no other way to make function work properly with first map parameter.

File: vars/longevityPipeline.groovy (hunk 1 above)
Comment (dkropachev) [reply vars/longevityPipeline.groovy:174]: We definitely need it in functions and therefore had to expose it, also we are currently exposing it via pipelineParam.

Comment (dkropachev) [reply vars/longevityPipeline.groovy:174]: @fruch, I will add test_name too.

Comment (dkropachev) [reply vars/longevityPipeline.groovy:174]: @fruch , test_name added

**Comment by bentsi** [comment]
@dkropachev Please check if we can add dynamic stages in post actions

**Comment by dkropachev** [comment]
> @dkropachev Please check if we can add dynamic stages in post actions

Unfortunately there is no way to have stages there, so I did move stages back to original place, but also added flags and if stage was not executed it is reexecuted in post action.
It gives middle solution, we still have stages if everything goes ok, and at the same time if pipeline is aborted and/or essential stage (cleanup resource, sending email, collecting logs) is not successfully executed during the run, we reexecute  it at the end anyway.

**Comment by bentsi** [comment]
@dkropachev what do you mean re-executed? will they run twice?

**Comment by dkropachev** [comment]
> @dkropachev what do you mean re-executed? will they run twice?

It step failed, it will be reexecuted at the end of pipeline, if not - it won't be.

**Comment by bentsi** [comment]
- don't retry "collect logs"
- try..except the sending email code

**Comment by dkropachev** [comment]
> * don't retry "collect logs"
> * try..except the sending email code

Done.

**Comment by bentsi** [comment]
did you try the new code on some test job?

**Comment by dkropachev** [comment]
> did you try the new code on some test job?

I did two abort tests: 1. Abort whole pipeline, abort test cycle
and three fail tests: fail all steps after checkout, fail sct run, fail send mail

**Pull request merged**